faststreams/inputs.nim

```diff
@@ -857,7 +901,7 @@ template readNImpl(sp: InputStream,
       startAddr = s.span.startAddr
       bumpPointer s.span, n
 
-  makeOpenArray(startAddr, n)
```

Comment:
The elimination of `makeOpenArray` is unattractive. What is gained from it?

Reply:
the difficulty when reviewing is that `makeOpenArray` _doesn't_ use the off-by-one ranges that nim normally employs so in the interest of keeping the code consistent with other nim code is to use `toOpenArray` throughout - my assumption was that this code was written before `toOpenArray` existed in a usable form

Reply:
`makeOpenArray` is less noisy and uses type inference to improve the type safety of the code.
I personally find the `(start, length)` semantics that it uses more aligned with the rest of the logic in FastStreams and thus easier to review.

Reply:
Another reason why the off-by-one argument doesn't carry enough weight for me is that you can consider `makeOpenArray` a constructor similar to `newSeq`. You don't't write `newSeq[T](n - 1)` after all.

Reply:
changed them back - there's now a healthy mix of makeOpenArray and toOpenArray